include/zephyr/init.h

```diff
@@ -15,150 +15,129 @@
 extern "C" {
 #endif
 
-/*
- * System initialization levels. The PRE_KERNEL_1 and PRE_KERNEL_2 levels are
- * executed in the kernel's initialization context, which uses the interrupt
- * stack. The remaining levels are executed in the kernel's main task.
+/**
+ * @defgroup sys_init System Initialization
+ *
+ * Zephyr offers an infrastructure to call initialization code before `main`.
+ * Such initialization calls can be registered using SYS_INIT() or
+ * SYS_INIT_NAMED() macros. By using a combination of initialization levels and
+ * priorities init sequence can be adjusted as needed. The available
+ * initialization levels are described, in order, below:
+ *
+ * - `PRE_KERNEL_1`: Executed in Kernel's initialization context, which uses
```

Comment:
what about `ARCH`?

Reply:
I will add it, apparently it was added without any documentation (sigh).

Reply:
added https://github.com/zephyrproject-rtos/zephyr/pull/51176